pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -32,67 +34,119 @@ import (
 )
 
 // ValidateKubeSchedulerConfiguration ensures validation of the KubeSchedulerConfiguration struct
-func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) field.ErrorList {
-	allErrs := field.ErrorList{}
-	allErrs = append(allErrs, componentbasevalidation.ValidateClientConnectionConfiguration(&cc.ClientConnection, field.NewPath("clientConnection"))...)
-	allErrs = append(allErrs, componentbasevalidation.ValidateLeaderElectionConfiguration(&cc.LeaderElection, field.NewPath("leaderElection"))...)
-
+func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) utilerrors.Aggregate {
+	var errs []error
+	errs = append(errs, componentbasevalidation.ValidateClientConnectionConfiguration(&cc.ClientConnection, field.NewPath("clientConnection")).ToAggregate())
+	errs = append(errs, componentbasevalidation.ValidateLeaderElectionConfiguration(&cc.LeaderElection, field.NewPath("leaderElection")).ToAggregate())
```

Comment:
@alculquicondor  This will bring us the extra cost to call `ToAggregate()` to convert the `ErrorList` to `Aggregate`, but the conversion itself seems useful when there are duplicated `error` in the  `ErrorList`

Reply:
And `go` doesn't allow us to do something like this, this seems the only way out if we don't want to mix `[]error` and `field.ErrorList` in this function.

```
errs = append(errs, componentbasevalidation.ValidateClientConnectionConfiguration(&cc.ClientConnection, field.NewPath("clientConnection"))...)
```

[1] https://golang.org/doc/faq#convert_slice_of_interface

Reply:
Yes, use `...`

Reply:
hmm... sorry, I cannot get it. 

I have explained why `...` is not applicable here, could you pls more explicit?

Reply:
I thought you were asking which one was preferred.
I understand why `...` doesn't work in this case now.

Reply:
my bad, I didn't express it clearly.